File: Algorithms/Sorting/bubblesort.cpp
#include <bits/stdc++.h>
#include <cstdlib>
#include <time.h>
using namespace std;
//bubble sort has time complexity O(n^2)
int main()
{
  srand(time(0));
  int length=rand()%10;
  int A[length];
  for(int i=0;i<length;i++)
  {
    A[i]=rand()%100;
  }
  int temp;
  int check;
  for(int i=0;i<length;i++)
  {
    check=1;
    for(int j=0;j<length-1;j++)
    {
      if(A[j]>A[j+1])
      {
        temp=A[j];
        A[j]=A[j+1];
        A[j+1]=temp;
        check=0;
      }
    }
    if(check) break;
  }
  for(int i=0;i<length;i++)
    cout<<A[i]<<',';
  cout<<'\b';  
}

File: Algorithms/Sorting/bubblesort.py
import random
A=random.sample(range(100),random.randint(1,10))
check=1
for x in A:
    check=1
    for y in range(len(A)-1):
        if A[y]>A[y+1]:
            temp=A[y]
            A[y]=A[y+1]
            A[y+1]=temp
            check=0
    if check:
        break
print(A)
